### PR TITLE
feat: support responsive design in modals

### PR DESCRIPTION
**Pull request by el**
### Describe what this PR does
- Support responsive designs for the modals

## Events
- [x] No new events

### On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [x] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [x] MobileL (420px)
- [x] Tablet (656px)
- [x] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-729 #done

### CODE REVIEW

File: packages/shared/src/components/modals/common/Modal.tsx [hunk 1]
```diff
@@ -13,6 +13,43 @@ export type ModalProps = ReactModal.Props & {
...
+> = {
+  [ModalKind.FlexibleTop]: {
+    [ModalSize.Medium]: 'mobileL:pt-10',
+  },
+};
+const modalKindToClassName: Record<ModalKind, string> = {
+  [ModalKind.FixedCenter]:
+    'h-full max-h-[calc(100vh-2.5rem)] mobileL:h-[40rem] mobileL:max-h-[calc(100vh-5rem)] mt-10 mobileL:mt-0',
+  [ModalKind.FlexibleCenter]:
+    'mx-4 max-w-[calc(100vw-2rem)] max-h-[min(calc(100vh),40rem)] mobileL:max-h-[min(calc(100vh-5rem),40rem)]',
```
Comment (rebelchris) [inline packages/shared/src/components/modals/common/Modal.tsx:32]: Nothing wrong with this, but when using vh and calc it's very important to solidly test it out on actual mobile devices.
I know safari on iPhone has some issues with vh.

File: packages/shared/src/components/modals/common/ModalBody.tsx [hunk 2]
```diff
@@ -1,26 +1,16 @@
...
-            kind === ModalKind.FlexibleTop && size === ModalSize.Large
-              ? 'p-8'
-              : 'p-6',
-          )}
-        >
-          {children}
-        </section>
-      )}
-    </ModalPropsContext.Consumer>
+  const { kind, size } = useContext(ModalPropsContext);
```
Comment (rebelchris) [inline packages/shared/src/components/modals/common/ModalBody.tsx:10]: Amazing, love this refactor ❤️

File: packages/shared/src/components/modals/common/ModalHeader.tsx [hunk 3]
```diff
@@ -1,18 +1,27 @@
...
 import { ModalPropsContext } from './types';
 
 export type ModalHeaderProps = {
-  title?: ReactNode;
+  children?: ReactNode;
+  title?: string;
 };
 
-export function ModalHeader({ title }: ModalHeaderProps): ReactElement {
+const ModalHeaderTitle = classed('h3', 'font-bold typo-title3');
```
Comment (rebelchris) [inline packages/shared/src/components/modals/common/ModalHeader.tsx:11]: I'm all for these, but why do we used it for the title, but not the header?
Perhaps we should find a uniform for this?

**Review by rebelchris — APPROVED** [review]
Looks great!
I would love to give this a proper test on actual devices though.

Doesn't block merging, but please let me know which modal I can eventually test.

File: packages/shared/src/components/modals/common/Modal.tsx (hunk 1 above)
Comment (el) [reply packages/shared/src/components/modals/common/Modal.tsx:32]: Yeap, makes perfect sense, I left the mobile device tests out of this. Let me add a jira just for this :
https://dailydotdev.atlassian.net/browse/WT-734

File: packages/shared/src/components/modals/common/ModalHeader.tsx (hunk 3 above)
Comment (el) [reply packages/shared/src/components/modals/common/ModalHeader.tsx:11]: I used for the header to be able to expose it outside of this component. Do we need it for everything?
My thought was to only use it for exposed components.

**Comment by el** [comment]
> Looks great! I would love to give this a proper test on actual devices though.
> 
> Doesn't block merging, but please let me know which modal I can eventually test.
@rebelchris 


After todays PR, 
Most of the Modals can be moved to the new structure. 
We can have a quick call about this 👍

**Pull request merged**